src/Core/Composer/ModulesConfigurationProcessor.php

```diff
@@ -0,0 +1,138 @@
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Composer;
+
+use Composer\IO\IOInterface;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+final class ModulesConfigurationProcessor
+{
+    /**
+     * @var IOInterface the CLI IO interface
+     */
+    private $io;
+
+    public function __construct(IOInterface $io)
+    {
+        $this->io = $io;
+    }
+
+    /**
+     * Foreach module, will install the module dependencies
+     * into /modules/{module}/vendor folder.
+     *
+     * @param array $configuration the Composer configuration
+     * @param string $rootPath the modules directory location
+     */
+    public function processInstallation(array $configuration, $rootPath)
+    {
+        $this->io->write('<info>PrestaShop Module installer</info>');
+
+        if (!array_key_exists('native-modules', $configuration) || !array_key_exists('modules-dir', $configuration)) {
+            return;
+        }
+
+        $nativeModules = $configuration['native-modules'];
+        $modulesLocation = $rootPath . $configuration['modules-dir'];
+
+        foreach ($nativeModules as $moduleName => $moduleVersion) {
+            $this->io->write(sprintf('<info>Looked into "%s" module (version %s)</info>', $moduleName, $moduleVersion));
+
+            $this->installModule($moduleName, $moduleVersion, $modulesLocation);
+        }
+    }
+
+
+    /**
+     * Foreach module, will install the module dependencies
+     * into /modules/{module}/vendor folder.
+     *
+     * @param array $configuration the Composer configuration
+     * @param string $rootPath the modules directory location
+     */
+    public function processUpdate(array $configuration, $rootPath)
+    {
+        $this->io->write('<info>PrestaShop Module installer</info>');
+
+        if (!array_key_exists('native-modules', $configuration) || !array_key_exists('modules-dir', $configuration)) {
+            return;
+        }
+
+        $nativeModules = $configuration['native-modules'];
+        $modulesLocation = $rootPath . $configuration['modules-dir'];
+
+        foreach ($nativeModules as $moduleName => $moduleVersion) {
+            $this->io->write(sprintf('<info>Looked into "%s" module (version %s)</info>', $moduleName, $moduleVersion));
+
+            $this->updateModule($moduleName, $moduleVersion, $modulesLocation);
+        }
+    }
+
+    private function installModule($moduleName, $moduleVersion, $location)
+    {
+        $moduleInformation = ModuleInformation::createFromString($moduleName, $moduleVersion);
+
+        if (file_exists($location . $moduleInformation->getName())) {
+            $this->io->write(sprintf('Module "%s" is already installed, skipped.', $moduleInformation->getName()));
+
+            return;
+        }
+
+        $command = 'composer create-project '.$moduleName.':'.$moduleVersion;
```

Comment:
hum, not agreed about using the `composer` command like that. What about composer.phar and different path which can be different between systems :/

Reply:
this is not an issue (ofc i will update it) because I've added composer/composer as PrestaShop dependencies.

So we have access to /vendor/composer/composer/bin/composer.

This is the one I will require in the end, not the one installed by the users.

Reply:
Updated

Reply:
@mickaelandrieu So you added composer as a composer dependency of the project ? composer requires composer ?! 🚀